docker-compose-net/docker-compose-net.md

```diff
@@ -0,0 +1,160 @@
+ 
+ **3.** A third method uses `weave run` commands to attach containers. This method also
+ does not use the Weave Docker API Proxy. 
+ 
+ See [Launching Containers With Weave Run (without the Proxy)](/site/weave-docker-api/launching-without-proxy.md).
+ 
+
+##<a name="setup-proxy"></a>Docker Compose and the Weave Net Docker API Proxy
+
+
+The following is a simple setup that shows how to configure docker-compose for Weave Net across two hosts using the Docker API proxy. You will launch the alpine using Docker Compose and then run 'ping' to test the connectivity between the two hosts. 
+
+**1.** Create two hosts on Virtualbox with docker machine: 
+
+~~~bash
+docker-machine create -d virtualbox weave-net-demo-01
+docker-machine create -d virtualbox weave-net-demo-02
+~~~
+
+**2.** In order to peer hosts on Weave Net, obtain the IP address of one of the VMs you just created: 
+
+~~~bash
+docker-machine ip weave-net-demo-02
+~~~
+
+Make a note of the IP address that appears. 
+
+**3.** Install and then launch Weave Net onto both VMs:
+
+~~~bash
+sudo curl -L git.io/weave -o /usr/local/bin/weave
+sudo chmod +x /usr/local/bin/weave
+weave launch <host-02-ip>
+~~~
+
+**4.** Setup the environment to run the Weave Docker API proxy: 
+
+~~~bash
+eval $(weave env)
+~~~
+
+With the Weave environment set up, you can run regular Docker commands on the same command line from which you created VMs. If you are attaching containers to the Weave Network, this step is mandatory, unless you've opted to use the Weave Net Docker Plugin which is described in part XX. 
+
+**5.** Run the docker compose file and direct each app to a host: 
+
+~~~bash
+cd proxy-compose
+docker-compose up -d pingme weave-net-demo-01 
+docker-compose up -d pinger weave-net-demo-02
+~~~
+
+###<a name="compose-proxy"></a>Docker Compose File for the Weave Docker API Proxy###
+
+The network_mode is set to "bridge" for this mode. 
+
+~~~bash
+version: '2'
+
+services:
+ ping:
+    image: alpine
+    container_name: pingme
+    command: nc -p 4000 -ll -e echo 'Hello, Weave!'
+    network_mode: "bridge"
+    
+ ping:
+    image: alpine
+    container_name: pinger
+    command: sh -l
+    hostname: hello.weave.local
+    network_mode: "bridge"
+~~~
+
+###<a name="compose-plugin"></a>Docker Compose File Variants and the Weave Net Docker Plugin
+
+If you are using the plugin, there are several different ways in which to setup your Docker Compose file. Which method you use, depends on your application's requirements: 
+
+1. Using "networks: - weave" and "dns: ..." and "dns_search: ..." service stanzas and "networks: weave: external: true" networks stanza. This uses the default 'weave' network for everything. It doesn't require a docker cluster and it uses weaveDNS.
+
+2. Using "networks: default: driver: weave". This places all containers on a user-defined network and it uses the 'weave' driver. It assumes that you have a Docker cluster. This method uses Docker's DNS.
+
```

Comment:
To use Docker DNS across multiple machines requires a cluster store, so rule this one out.